src/MatrixTypes.f90

```diff
@@ -64,337 +65,49 @@ MODULE MatrixTypes
   USE Allocs
   USE BLAS2,           ONLY: BLAS2_matvec => BLAS_matvec
   USE BLAS3,           ONLY: BLAS3_matmult => BLAS_matmat
-  USE trilinos_interfaces
   USE ParameterLists
   USE VectorTypes
+  USE MatrixTypes_Base
+  USE MatrixTypes_Native
+  USE MatrixTypes_Trilinos
+  USE MatrixTypes_PETSc
+  USE trilinos_interfaces
   IMPLICIT NONE
 
-#ifdef FUTILITY_HAVE_PETSC
-#include <finclude/petsc.h>
-#undef IS
-#endif
-
   PRIVATE
 !
 ! List of public members
-  PUBLIC :: eMatrixType
   PUBLIC :: MatrixType
+  PUBLIC :: MatrixFactory
+  PUBLIC :: DistributedMatrixFactory
+  PUBLIC :: MatrixResemble
   PUBLIC :: SquareMatrixType
   PUBLIC :: RectMatrixType
+  PUBLIC :: DistributedMatrixType
+  ! Matrix structure enumerations
+  PUBLIC :: SPARSE,DENSESQUARE,DENSERECT,TRIDIAG
+  ! Matrix-Vector engine enumerations
+  PUBLIC :: VM_PETSC,VM_TRILINOS,VM_NATIVE
```

Comment:
To clarify, I was worried about cluttering the namespace with overly-general NATIVE, TRILINOS, PETSC enumerators.